### PR TITLE
Allow Long as ID type in Entity for Cosmos DB

### PR DESCRIPTION
**Pull request by moarychan**

### CODE REVIEW

**Comment by moarychan** [comment]
Related issue:
https://github.com/Azure/azure-sdk-for-java/issues/12712

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java [hunk 1]
```diff
@@ -454,9 +455,20 @@ public void deleteById(String containerName, Object id, PartitionKey partitionKe
         Assert.notNull(ids, "Id list should not be null");
         Assert.notNull(domainType, "domainType should not be null.");
         Assert.hasText(containerName, "container should not be null, empty or only whitespaces");
-
+        final List<Object> idList = new ArrayList<>();
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java:458]: I think we need this change for `ReactiveCosmosTemplate` as well.

File: sdk/cosmos/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/repository/repository/LongIdDomainRepository.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.data.cosmos.repository.repository;
+
+import com.azure.spring.data.cosmos.domain.LongIdDomain;
+import com.azure.spring.data.cosmos.repository.CosmosRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface LongIdDomainRepository extends CosmosRepository<LongIdDomain, Long> {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/repository/repository/LongIdDomainRepository.java:10]: Please make sure this also works for `ReactiveCosmosRepository`

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/CosmosEntityInformation.java [hunk 3]
```diff
@@ -201,8 +201,10 @@ private Field getIdField(Class<?> domainType) {
             throw new IllegalArgumentException("domain should contain @Id field or field named id");
         } else if (idField.getType() != String.class
                 && idField.getType() != Integer.class
-                && idField.getType() != int.class) {
-            throw new IllegalArgumentException("type of id field must be String or Integer");
+                && idField.getType() != int.class
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/CosmosEntityInformation.java:204]: Can we add a unit test for this in `CosmosEntityInformationUnitTest.java` class ?

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java (hunk 1 above)
Comment (moarychan) [reply sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java:455]: Should be, but there's o similar method in ReactiveCosmosTemplate,  I unified the procesing of ID type conversion.

File: sdk/cosmos/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/repository/repository/LongIdDomainRepository.java (hunk 2 above)
Comment (moarychan) [reply sdk/cosmos/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/repository/repository/LongIdDomainRepository.java:10]: added  ReactiveLongIdDomainPartitionPartitionRepositoryIT, please review

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/CosmosEntityInformation.java (hunk 3 above)
Comment (moarychan) [reply sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/CosmosEntityInformation.java:204]: Added in CosmosEntityInformationUnitTest, please check

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/common/CosmosUtils.java [hunk 4]
```diff
@@ -45,4 +47,26 @@
...
+
+    /**
+     * ID value should be string value, real id type will be String, Integer, Long,
+     * all of these must be converted to String type.
+     * @param idValue id value to find
+     * @throws IllegalArgumentException thrown if id value fail the validation.
+     * @throws IllegalQueryException thrown if id value fail the validation.
+     * @return String id value
+     */
+    public static Object getStringIDValue(Object idValue) {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/common/CosmosUtils.java:59]: This should return <String> type, why is it returning Object ?

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java (hunk 1 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java:454]: Good, makes sense.

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/CosmosEntityInformation.java (hunk 3 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/CosmosEntityInformation.java:204]: Thanks.

File: sdk/cosmos/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/repository/repository/LongIdDomainRepository.java (hunk 2 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/repository/repository/LongIdDomainRepository.java:10]: Looks good, thanks.

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/common/CosmosUtils.java (hunk 4 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/common/CosmosUtils.java]: @moarychan - I have pushed a commit to fix this.

**Pull request merged**